### ISSUE TITLE
GameGuru MAX - Material "Flickering" and Light Probe Issue

### ISSUE THREAD

**Issue by ArgentArts** [TheGameCreators/GameGuruRepo#2246]
I notice that shiny materials "flicker" or change depending on camera/player distance. I am guessing this is due to light probes? Here is a link to a video showing this (I recommend right-click and download to watch):

http://www.argentarts.com/uploads/flicker.mp4

Notice at first the shiny ball in the center (with the squares on it) - it's almost black at first. Then, after it sort of snaps in, it never goes black again. But as the camera moves in and back out again, you can see changes to it that appear to be from different light probes. Currently, this area has two point lights in it. But I had also tested this with only one light and the same amount of changes take place on the shiny spheres, as if there are more than one light probe. In fact, when I selected the point light and went to adjust the light probe side, more than one light probe showed up in the scene, NEITHER located where the point light was?!?

Light probes should not be connected to lights. They should be independent items that the end-user sets up themselves. The end-user should be able to set up any number of lights of various kinds in an area and then set up a single light probe in that one area to hand reflections, etc. The end-user should be able to select and position a light-probe independent of any light source.

Speaking of Light Probes - As of the current release, I can no longer see the light probe size when resizing. In other words, when I drag the light probe size slider, I don't see the sizing cube I used to see in the view port. Instead, I see a silver sphere show up that disappears when I release the slider. This sphere does not appear near the point light, btw. In the following screen shot, TWO light probes are showing up even though there's only one point light:

![light_probes](https://user-images.githubusercontent.com/36735507/155260483-6a901d84-87a0-48fa-833a-8485d202e1c3.png)

I believe this is because I had one point light there previously to test this, had deleted it, and then had added another one to take this screen shot. If this is the case, then the light probes are being left behind.

**Comment by ArgentArts** [comment]
There are several issues here. Firstly, the bug is that you cannot see the light probe area any more when you go to adjust it. So, you cannot set the size visually for your areas now. But, to me, **the larger concern is that light probes should not be attached to point lights at all, but should be their own objects.** If I have four point lights to light a room, I don't need four light probes. And, as seen from my video, it can cause issues with reflecting objects as the light probes will compete with each other depending on camera position. This causes huge visual issues - popping that is very visually distracting.

Ultimately, we really need to separate light probes from point lights and make them their own thing, allowing the end-user to place them strategically in rooms only where needed (and they are not always or frequently needed).

**Comment by PaulSJ** [comment]
Light probes used to be connected to lights but that is no longer the case. Probes are now connected to the player, which is why things are changing as you move around. I'll adjust the transition time so the changes are more subtle.

It would be great if we could have the user place light probes all over the level but WickedEngine is limited to 16 probes, which isn't really enough for a large level

**Comment by ArgentArts** [comment]
Also, if light probes are not connected to the player, does that mean the light probe scale is going to disappear from the UI for lights?

**Comment by PaulSJ** [comment]
The PBR pipeline uses light probes for diffuse lighting as well as specular, so they are technically required for everything

> does that mean the light probe scale is going to disappear from the UI for lights?
I would assume so but I've not heard it discussed

**Comment by ArgentArts** [comment]
Ah! Yes. I am thinking very old-school here. Thanks for clearing that up. I appreciate it.

**Comment by ArgentArts** [comment]
Yeah, it seems sort of useless to have a light probe scale setting that does nothing for each light and it will be sure to confuse the end-user. I think it's best to remove it from the UI before the March release, imo.

**Comment by ArgentArts** [comment]
> I'll adjust the transition time so the changes are more subtle.

Can the transition distance be made greater? In the above scene, the player is not moving very far and the probe is changing two or three times. That seems excessive.

**Comment by PaulSJ** [comment]
The problem with making the change distance greater is that when the player enters a building the environment probe might still be outside, which would make the indoor lighting incorrect. It's possible we could try and detect transitions from indoor to outdoor, and room to room, but for now I think we'll have to live with the current system.

**Comment by ArgentArts** [comment]
What about for people like me, who are creating in-door only games? I don't need to worry about such transitions? Couldn't there be a toggle for something like this? As can be seen in my video, the transition in reflective surfaces is very jarring? 

I hope you don't mind the questions. I'm just exploring. 

Ultimately, it would be awesome if the end-user could set their own distance (within reason) - the best of both worlds.

**Comment by ArgentArts** [comment]
Here's an issue where I am pretty close to a reflective object, but the object appears to be using an incorrect light probe:

![pbr6](https://user-images.githubusercontent.com/36735507/155385832-b2a728db-4897-4e8b-802f-96becd6dd836.png)

Notice the green reflection to the left side of the reflective sphere? It should be reflecting the rock brick sphere that's next to it. Instead, it's picking up a green sphere that's more toward the middle of the room (left side middle of the room). So, the transition has yet to take place even though the player and camera are quite close to the sphere.

**Comment by PaulSJ** [comment]
> the transition in reflective surfaces is very jarring

This should be better in the next build, the transition time will be slower

> Notice the green reflection to the left side of the reflective sphere? It should be reflecting the rock brick sphere that's next to it

This will be a problem unless the light probe is directly centered on the reflective sphere. To get it perfect every reflective object would need to have a light probe for itself, not impossible but it would use a lot of light probes 

> We keep the Light Probe for the player. But we also have a separate object called a Light Probe that is limited to a specific number 15

I think there does need to be a manual light probe system for those users that want to use it, but it would be a big change so we'll have to look at it after the initial release

**Comment by ArgentArts** [comment]
Definitely was not expecting something before the March release. But it would be nice to see down-the-road, for certain. As an artist, lighting is one of the most important things you can work on in a game, especially for indoor levels. When an artist does not have any control (or very limited control) over light probes, it takes away a lot from the artist ... it removes or extremely limits one of the artists main tools for bringing their creations to life. This is why I "battle" for things like this and point light shadows so hard. :)

**Comment by MonkeyFrogStudio** [comment]
Since Wicked is limited to 16 light probes, instead of having one for the player, why not treat them like we do point and spotlights that cast shadows? Only the NEAREST works. Any beyond a certain range are disabled, only enabled when the player is within range? This way, the end-user could place as many light probes as they please, but only no more than 16 could ever be active and only those that are nearest the player.

**Comment by synchromesh62** [comment]
> Since Wicked is limited to 16 light probes, instead of having one for the player, why not treat them like we do point and spotlights that cast shadows? Only the NEAREST works. Any beyond a certain range are disabled, only enabled when the player is within range? This way, the end-user could place as many light probes as they please, but only no more than 16 could ever be active and only those that are nearest the player.

we had that originally with 5 lights ... just cause flickering in the distance etc.
The option we have now to enable or disable specific lights to cast shadows is better as you can strategically place them
Unlesss your thinking of some kind of fading in process or im misreading.

**Comment by MonkeyFrogStudio** [comment]
@synchromesh62 I am not at all talking about lights here. I am talking about LIGHT PROBES.

**Comment by synchromesh62** [comment]
> @synchromesh62 I am not at all talking about lights here. I am talking about LIGHT PROBES.

Whoops .. Doh !!
Sorry :)

**Comment by MonkeyFrogStudio** [comment]
No worries. ;)

**Comment by davetgc** [comment]
This and similar issues is being reported by quite a few users as a show stopper, and they all seem to stem from the light probes. It's far more obvious in interior locations and I tend to agree that this can really jar visually.

**Comment by LeeBamberTGC** [comment]
@MonkeyFrogStudio Now added the ability in the advanced settings of the light properties to specify a light as a light probe and adjust the range for each of the probes. The new probe system will use these virtual probes to assign eight real probes as the player moves through the level. Careful placement of probes can now produce correct PBR visuals for the lighted scene.

**Comment by MonkeyFrogStudio** [comment]
I'm closing this one due to the new light/environment probes in MAX.
